lib/dry/struct/class_interface.rb

```diff
@@ -52,16 +52,17 @@ def check_schema_duplication(new_schema)
       private :check_schema_duplication
 
       def constructor_type(type = nil)
-        if type
-          @constructor_type = type
-        else
-          @constructor_type || :strict
-        end
+        return @constructor_type = type if type
+        return @constructor_type = :strict unless defined? @constructor_type
+
+        @constructor_type
```

Comment:
Both situations are not valid. Bacuase if they fail test when constructir_type is set in parent and then should be same in inherited 'child' class